tests/Feature/LangTest.php

```diff
@@ -35,12 +41,48 @@ public function testLang(): void
 			$locale = $lang_test->get_locale();
 
 			foreach ($keys as $key) {
-				static::assertArrayHasKey($key, $locale, 'Language ' . $lang_test->code() . ' is incomplete.');
+				try {
+					static::assertArrayHasKey($key, $locale, 'Language ' . $lang_test->code() . ' is incomplete.');
+				} catch (ExpectationFailedException $e) {
+					$msgSection->writeln('<comment>Error:</comment> ' . $e->getMessage());
+					$failed = true;
+				}
 			}
 		}
 
 		static::assertEquals('en', Lang::get_code());
 		static::assertEquals('OK', Lang::get('SUCCESS'));
+		$this->assertFalse($failed);
+	}
+
+	/**
+	 * Test Languages are strictly.
+	 *
+	 * @return void
+	 */
+	public function testLangOverflow(): void
+	{
+		$msgSection = (new ConsoleOutput())->section();
+
+		$failed = false;
+
+		$full = Lang::get_lang();
+		$lang_available = Lang::get_lang_available();
+
+		foreach ($lang_available as $code) {
+			$lang_test = Lang::factory()->make($code);
+			$keys = array_keys($lang_test->get_locale());
+
+			foreach ($keys as $key) {
+				try {
+					static::assertArrayHasKey($key, $full, 'Language ' . $lang_test->code() . ' as too many keys.');
+				} catch (ExpectationFailedException $e) {
+					$msgSection->writeln('<comment>Error:</comment> ' . $e->getMessage());
+					$failed = true;
+				}
```

Comment:
Instead of looping across all the keys we could simply use the PHP native method [`array_diff_key`](https://www.php.net/manual/en/function.array-diff-key.php). It should make the code a little bit more efficient (not that I care such much about it in the test), but it should also make the code much more clearer and easier to grasp.

Moreover, this test method and the test method above could be combined into one. Basically, they do the same thing just with swapped roles. So that is the pseudo code:

```php
public function testLanguageConsistency(): void {
	$englishDictionary = Lang::get_lang();
	$availableDictionaries = Lang::get_lang_available();
	$failed = false;
	foreach ($availableDictionaries as $locale) {
		$dictionary = Lang::factory()->make($locale);
		$missingKeys = array_diff_key(englishDictionary, $dictionary);
		$extraKeys = array_diff_key($dictionary, englishDictionary);
		if (size($missingKeys) !== 0) {
			$msgSection->writeln(sprintf('<comment>Error:</comment> Locale %s misses the following keys: %s', $locale, implode($missingKeys, ', ')));
			$falied = true;
		}
		if (size($extraKeys) !== 0) {
			$msgSection->writeln(sprintf('<comment>Error:</comment> Locale %s has the following extra keys: %s', $locale, implode($extraKeys, ', ')));
			$falied = true;
		}
	}
	static::assertFalse($failed);
}
```